packages/dds/merge-tree/src/localReference.ts

```diff
@@ -305,7 +304,7 @@ export class LocalReferenceCollection {
             node?.list?.remove(node);
 
             lref.link(
-                undefined,
+                lref.getSegment(),
```

Comment:
ugggh. i hate this right now. i changed it to undefined in a previous pr  #11778  , as i think its the correct behavior, otherwise a removed ref === a transient ref, which i think is wrong. However, here i run into reentrancy problems with link. they are fixable, but the fixes are gross. We only need the reentrancy to update the count here (ref, heir, tile), which i also think is wrong, and would like to remove, if possible, which i think it is.  I'll open an issue or two in there area to get it cleaned up.

Reply:
Sounds good to me.

Reply:
AB#1934